v201705/ad_group_ad.go

```diff
@@ -13,6 +14,12 @@ type AppUrl struct {
 	OsType string `xml:"osType"` // "OS_TYPE_IOS", "OS_TYPE_ANDROID", "UNKNOWN"
 }
 
+type Ad struct {
```

Comment:
What's the purpose of this struct?

Reply:
I made this change because adwords was expecting type Ad in the xml while updating ExpandedTextAds, (Ad is an interface type which adwords expects). So I created a struct Ad with only fields which can be modified for text ads and added another case in switch case for type Ad in the marshalling portion of ad_group_ad.go of gads

Reply:
It might make sense to put this struct in a separate file for text ads instead of lumping it in all ads.